src/algorithms/transfermatrix.jl

```diff
@@ -0,0 +1,38 @@
+#
+# Edge transfer matrices
+#
+
+# single site transfer
+struct EdgeTransferMatrix{A <: CTMRGEdgeTensor, B, C <: CTMRGEdgeTensor} <: MPSKit.AbstractTransferMatrix
+    top::A
+    mid::B
+    bot::C
+    isflipped::Bool
+end
```

Comment:
This needs a decision on the name, and whether it's worth it so subtype `MPSKit.AbstractTransferMatrix`. The upside is that we get some free convenience syntax regarding products of and multiplication with transfer matrices. The downside is that there's no way to dispatch to this instead of `MPSKit.SingleTransferMatrix` based on the arguments, which means that we need an explicit constructor (with yet another name) in the end anyway (see below).